MediaRecorder.bs

```diff
@@ -286,11 +276,35 @@ interface MediaRecorder : EventTarget {
     <li>Set |recorder|'s {{state}} to {{recording}}, and run the following steps
     in parallel:
     <ol>
-      <li>Start recording all tracks in |tracks| using the {{MediaRecorder}}'s
-      current configuration, and gather the data into a {{Blob}} <var>blob</var>
-      and queue a task, using the DOM manipulation task source, to
-      <a>fire an event</a> named <a>start</a> at <var>recorder</var>, then return
-      <code>undefined</code>.</li>
+      <li>Let the User Agent specify the |recorder|'s configuration for
+      recording all the tracks in |tracks|, given its current constraints. If
+      track data is available, the User Agent MAY take the content into account
+      when making this decision as long as this does not violate how the
+      configuration has previously been constrained (see earlier steps).</li>
+      <div class="note">
+        By looking at the content of the tracks when recording start, the User
+        Agent could take frame information into account. For example, if the
+        MediaStreamTrack is a remote track sourced from an RTCPeerConnection,
+        the User Agent is allowed to pick the same codec that the source RTP
+        stream uses, which may allow the User Agent to avoid the cost of
+        re-encoding by recording the raw encoded frames.
+      </div>
+      <li>Start recording all tracks in |tracks| using the |recorder|'s current
+      configuration and gather the data into a {{Blob}} <var>blob</var>. As soon
+      as recording starts, queue a task, using the DOM manipulation task source,
+      to run the following steps:
+      <ol>
+        <li>Let |extendedMimeType| be the value of |recorder|'s
+        [=[[ConstrainedMimeType]]=] slot.</li>
+        <li>Modify |extendedMimeType| by adding media type, subtype and codecs
+        parameter reflecting the configuration used by the MediaRecorder to
+        record all tracks in |tracks|, if not already present. This MAY include
+        the profiles parameter [[RFC6381]] or further codec-specific
+        parameters.</li>
+        <li>Set |recorder|'s {{mimeType}} attribute to |extendedMimeType|.</li>
+        <li><a>Fire an event</a> named <a>start</a> at |recorder|.</li>
```

Comment:
This language as written allows the MIME type to be modified "as soon as recording starts", not just to be "set if unset". Is that intentional?

Reply:
Yes mimeType is modified to match the configuration. This is the same text as prior to this PR, except it was moved to "onstart" instead of happening prior to starting encoding at start().